### DISCUSSION TITLE
ESLint error from the installed config/auth.ts file

### DISCUSSION THREAD

**Discussion by gregmsanderson** [adonisjs/core#4716] [General]
Hi,

A tiny thing which but out of interest I just tried the steps on https://adonisjs.com/blog/upgrading-eslint-typescript-setup to upgrade ESLint. That worked :) However when running the lint command I noticed an error:

```
/config/auth.ts
  24:13  error  'Authenticators' is already declared in the upper scope on line 3 column 32  @typescript-eslint/no-shadow

✖ 1 problem (1 error, 0 warnings)

 ELIFECYCLE  Command failed with exit code 1.
```

That `config/auth.ts` is the default file created when opting to use access tokens as the guard.

```js
import { defineConfig } from '@adonisjs/auth'
import { tokensGuard, tokensUserProvider } from '@adonisjs/auth/access_tokens'
import type { InferAuthEvents, Authenticators } from '@adonisjs/auth/types'

const authConfig = defineConfig({
  default: 'api',
  guards: {
    api: tokensGuard({
      provider: tokensUserProvider({
        tokens: 'accessTokens',
        model: () => import('#models/user'),
      }),
    }),
  },
})

export default authConfig

/**
 * Inferring types from the configured auth
 * guards.
 */
declare module '@adonisjs/auth/types' {
  interface Authenticators extends InferAuthenticators<typeof authConfig> {}
}
declare module '@adonisjs/core/types' {
  interface EventsList extends InferAuthEvents<Authenticators> {}
}
```

That file appears to have been generated from this:

https://github.com/adonisjs/presets/blob/b8a44f09332136c1e428ab43d6588982c41467c6/src/auth/stubs/config/access_tokens_with_lucid.stub#L27

Of course the ESLint error can be disabled with:

```js
declare module '@adonisjs/auth/types' {
  /* eslint-disable @typescript-eslint/no-shadow */
  interface Authenticators extends InferAuthenticators<typeof authConfig> {}
}
```

... but it would perhaps be _better_ to fix it. It looks to be caused by `Authenticators` being used twice. Are other people getting that? Ignoring it?

**Comment by thetutlage** [comment]
Thanks for reporting it. It has been fixed in this commit and will be part of next release. https://github.com/adonisjs/presets/commit/d15e72f157af12083610f9b73768f60c7a70ff88
